src/server/game/DungeonFinding/LFGMgr.cpp

```diff
@@ -1603,6 +1603,26 @@ namespace lfg
 
         _SaveToDB(gguid);
 
+        // Select a player inside to be teleported to
+        WorldLocation const* teleportLocation = nullptr;
+        for (GroupReference* itr = grp->GetFirstMember(); itr != nullptr; itr = itr->next())
+        {
+            Player* plr = itr->GetSource();
+            if (plr && plr->GetMapId() == uint32(dungeon->map))
+            {
+                // Remove bind to that map
+                if (!sInstanceSaveMgr->PlayerIsPermBoundToInstance(plr->GetGUID(), dungeon->map, plr->GetDungeonDifficulty()))
```

Comment:
```c++

        WorldLocation const* teleportLocation = nullptr;
+        Player* leader = ObjectAccessor::FindPlayerInOrOutOfWorld(grp->GetLeaderGUID());
        for (GroupReference* itr = grp->GetFirstMember(); itr != nullptr; itr = itr->next())
        {
            Player* plr = itr->GetSource();
            if (plr && plr->GetMapId() == uint32(dungeon->map))
            {
+                bool UnbindInstance = leader && plr->GetInstanceId() != leader->GetInstanceId();
                // Remove bind to that map
-                if (!sInstanceSaveMgr->PlayerIsPermBoundToInstance(plr->GetGUID(), dungeon->map, plr->GetDungeonDifficulty()))
+                if (UnbindInstance && !sInstanceSaveMgr->PlayerIsPermBoundToInstance(plr->GetGUID(), dungeon->map, plr->GetDungeonDifficulty()))


```

Reply:
Fix.
After the player escapes from the dungeon, the new player who joins the dungeon again is not in the same instance as the dungeon player..

Reply:
So with this fix the players' binds are never cleared, so always been on different instance id. `plr->GetInstanceId()` is always the same as `leader->GetInstanceId()` if being on the same.

Reply:
Well,  but it must be wrong not to be in the same instance

Reply:
I tested that and never encountered such a problem. As players has been teleported to the same instance. Can you tell me what to do to trigger that bug (step by step)?

Reply:
1. One player leaves the random team
2. New players join the random queue, enter the random dungeon, and then can't see other players

Reply:
Another problem is that if the last "new player" binds this instance before joining the random dungeon, the dungeon Finder will not add the "new player" to the instance of this dungeon

Reply:
@pklloveyou Should be fixed.